### PR TITLE
add --output-depth functionality on esgmapfile

### PR DESCRIPTION
**Pull request by alaniwi**
Hi, @ltroussellier 

I am creating this pull request to add a feature that lets you specify a depth that is the number of DRS elements below the top-level output directory when creating mapfiles.

For example, if you have dataset ID:
```
MIP-DRS7.CMIP7.CMIP.UKNCSP.UKCM2-0-LL.piControl.r1i1p1f1.glb.mon.abs550aer.tavg-u-hxy-u.g110.v20260729
```
and you specify `--outdir <topdir> --output-depth 5` (for some value of `<topdir>`), then the mapfile is stored at
```
<topdir>/MIP-DRS7/CMIP7/CMIP/UKNCSP/UKCM2-0-LL/MIP-DRS7.CMIP7.CMIP.UKNCSP.UKCM2-0-LL.piControl.r1i1p1f1.glb.mon.abs550aer.tavg-u-hxy-u.g110.v20260729.map
```
i.e. in this example it has created 5 levels of subdirectories.

We really need this feature at CEDA in order to organise the mapfiles in a tidy way and not dump everything into a flat directory.

I originally created the branch `mapfile_output_depth`, but it has not been merged (I think I forgot to create a PR), and now the branches have diverged a bit, so I have created a separate branch `mapfile_output_depth_new` with the commit rebased onto the current `main`.

I would be grateful if you can please merge this.

Thanks.

### CODE REVIEW

File: esgprep/mapfile/make.py [hunk 1]
```diff
@@ -109,13 +110,9 @@ def __call__(self, source):
...
-            # try:
-            #     outdir = outdir.joinpath(self.cfg.get(section='config:{}'.format(get_project(source)),
-            #                                           option='mapfile_drs',
-            #                                           vars=get_terms(source)))
-            #
-            # except:
-            #     pass
+
+            # Add subdirectories below outdir
+            outdir = outdir.joinpath(*outfile.name.split(".")[:self.output_depth])
```
Comment (tkettenba) [inline esgprep/mapfile/make.py]: Can this become an empty stringl, eg. since output_depth defaults to '0'.

**Comment by ltroussellier** [comment]
Thanks @alaniwi, looking into this now. Three things to sort out first:

1. **Guard `--output-depth` values.** Negative values nest every facet; values above the facet count leak the internal `.part` working extension as a directory (`.../v20260918/part/...`).
2. **Propagate to `esgmapfile show`.** It currently rejects the flag and reports the flat path, so it no longer matches what `make` writes.
3. **Add a test** covering the depth behaviour — default, a normal depth, and the edge cases above.

FYI on @tkettenba's point: depth `0` is fine — `[:0]` gives an empty list, so the default is unchanged.
